<a href="https://colab.research.google.com/github/Roja-thapa/Ecommerce_sales_analysis/blob/main/E_commerce.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [7]:
!pip install --upgrade plotly

In [8]:
import pandas as pd
import plotly.express as px
import plotly.graph_objects as go
import plotly.io as pio
import plotly.colors as colors
pio.templates.default= "plotly_white"

In [9]:
df= pd.read_csv("Sample - Superstore.csv",encoding='latin-1')
df.head()

Row ID        Order ID  Order Date   Ship Date       Ship Mode Customer ID  \
0       1  CA-2016-152156   11/8/2016  11/11/2016    Second Class    CG-12520   
1       2  CA-2016-152156   11/8/2016  11/11/2016    Second Class    CG-12520   
2       3  CA-2016-138688   6/12/2016   6/16/2016    Second Class    DV-13045   
3       4  US-2015-108966  10/11/2015  10/18/2015  Standard Class    SO-20335   
4       5  US-2015-108966  10/11/2015  10/18/2015  Standard Class    SO-20335   

     Customer Name    Segment        Country             City  ...  \
0      Claire Gute   Consumer  United States        Henderson  ...   
1      Claire Gute   Consumer  United States        Henderson  ...   
2  Darrin Van Huff  Corporate  United States      Los Angeles  ...   
3   Sean O'Donnell   Consumer  United States  Fort Lauderdale  ...   
4   Sean O'Donnell   Consumer  United States  Fort Lauderdale  ...   

  Postal Code  Region       Product ID         Category Sub-Category  \
0       42420   South  FUR-BO-10001798        Furniture    Bookcases   
1       42420   South  FUR-CH-10000454        Furniture       Chairs   
2       90036    West  OFF-LA-10000240  Office Supplies       Labels   
3       33311   South  FUR-TA-10000577        Furniture       Tables   
4       33311   South  OFF-ST-10000760  Office Supplies      Storage   

                                        Product Name     Sales  Quantity  \
0                  Bush Somerset Collection Bookcase  261.9600         2   
1  Hon Deluxe Fabric Upholstered Stacking Chairs,...  731.9400         3   
2  Self-Adhesive Address Labels for Typewriters b...   14.6200         2   
3      Bretford CR4500 Series Slim Rectangular Table  957.5775         5   
4                     Eldon Fold 'N Roll Cart System   22.3680         2   

   Discount    Profit  
0      0.00   41.9136  
1      0.00  219.5820  
2      0.00    6.8714  
3      0.45 -383.0310  
4      0.20    2.5164  

[5 rows x 21 columns]

In [10]:
df.isnull().sum()

Row ID           0
Order ID         0
Order Date       0
Ship Date        0
Ship Mode        0
Customer ID      0
Customer Name    0
Segment          0
Country          0
City             0
State            0
Postal Code      0
Region           0
Product ID       0
Category         0
Sub-Category     0
Product Name     0
Sales            0
Quantity         0
Discount         0
Profit           0
dtype: int64

In [11]:
df.duplicated().any()

np.False_

In [12]:
df.describe()

Row ID   Postal Code         Sales     Quantity     Discount  \
count  9994.000000   9994.000000   9994.000000  9994.000000  9994.000000   
mean   4997.500000  55190.379428    229.858001     3.789574     0.156203   
std    2885.163629  32063.693350    623.245101     2.225110     0.206452   
min       1.000000   1040.000000      0.444000     1.000000     0.000000   
25%    2499.250000  23223.000000     17.280000     2.000000     0.000000   
50%    4997.500000  56430.500000     54.490000     3.000000     0.200000   
75%    7495.750000  90008.000000    209.940000     5.000000     0.200000   
max    9994.000000  99301.000000  22638.480000    14.000000     0.800000   

            Profit  
count  9994.000000  
mean     28.656896  
std     234.260108  
min   -6599.978000  
25%       1.728750  
50%       8.666500  
75%      29.364000  
max    8399.976000

In [13]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9994 entries, 0 to 9993
Data columns (total 21 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Row ID         9994 non-null   int64  
 1   Order ID       9994 non-null   object 
 2   Order Date     9994 non-null   object 
 3   Ship Date      9994 non-null   object 
 4   Ship Mode      9994 non-null   object 
 5   Customer ID    9994 non-null   object 
 6   Customer Name  9994 non-null   object 
 7   Segment        9994 non-null   object 
 8   Country        9994 non-null   object 
 9   City           9994 non-null   object 
 10  State          9994 non-null   object 
 11  Postal Code    9994 non-null   int64  
 12  Region         9994 non-null   object 
 13  Product ID     9994 non-null   object 
 14  Category       9994 non-null   object 
 15  Sub-Category   9994 non-null   object 
 16  Product Name   9994 non-null   object 
 17  Sales          9994 non-null   float64
 18  Quantity

**Order date and ship date are in object datatype rather than date format.Therefore we need to change it's data type **

**CHANGE DATA TYPE FROM OBJECT TO DATETIME FORMAT**

In [14]:
df['Order Date']=pd.to_datetime(df['Order Date'])
df['Ship Date']=pd.to_datetime(df['Ship Date'])
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9994 entries, 0 to 9993
Data columns (total 21 columns):
 #   Column         Non-Null Count  Dtype         
---  ------         --------------  -----         
 0   Row ID         9994 non-null   int64         
 1   Order ID       9994 non-null   object        
 2   Order Date     9994 non-null   datetime64[ns]
 3   Ship Date      9994 non-null   datetime64[ns]
 4   Ship Mode      9994 non-null   object        
 5   Customer ID    9994 non-null   object        
 6   Customer Name  9994 non-null   object        
 7   Segment        9994 non-null   object        
 8   Country        9994 non-null   object        
 9   City           9994 non-null   object        
 10  State          9994 non-null   object        
 11  Postal Code    9994 non-null   int64         
 12  Region         9994 non-null   object        
 13  Product ID     9994 non-null   object        
 14  Category       9994 non-null   object        
 15  Sub-Category   9994 n

**ADDING NEW COLUMNS**

In [15]:
df['order month']=df['Order Date'].dt.month
df['order month_name']=df['Order Date'].dt.strftime('%B')  # monthname
df['order year']=df['Order Date'].dt.year
df['order Day of week']=df['Order Date'].dt.dayofweek
df.head()

Row ID        Order ID Order Date  Ship Date       Ship Mode Customer ID  \
0       1  CA-2016-152156 2016-11-08 2016-11-11    Second Class    CG-12520   
1       2  CA-2016-152156 2016-11-08 2016-11-11    Second Class    CG-12520   
2       3  CA-2016-138688 2016-06-12 2016-06-16    Second Class    DV-13045   
3       4  US-2015-108966 2015-10-11 2015-10-18  Standard Class    SO-20335   
4       5  US-2015-108966 2015-10-11 2015-10-18  Standard Class    SO-20335   

     Customer Name    Segment        Country             City  ...  \
0      Claire Gute   Consumer  United States        Henderson  ...   
1      Claire Gute   Consumer  United States        Henderson  ...   
2  Darrin Van Huff  Corporate  United States      Los Angeles  ...   
3   Sean O'Donnell   Consumer  United States  Fort Lauderdale  ...   
4   Sean O'Donnell   Consumer  United States  Fort Lauderdale  ...   

  Sub-Category                                       Product Name     Sales  \
0    Bookcases                  Bush Somerset Collection Bookcase  261.9600   
1       Chairs  Hon Deluxe Fabric Upholstered Stacking Chairs,...  731.9400   
2       Labels  Self-Adhesive Address Labels for Typewriters b...   14.6200   
3       Tables      Bretford CR4500 Series Slim Rectangular Table  957.5775   
4      Storage                     Eldon Fold 'N Roll Cart System   22.3680   

  Quantity Discount    Profit order month  order month_name  order year  \
0        2     0.00   41.9136          11          November        2016   
1        3     0.00  219.5820          11          November        2016   
2        2     0.00    6.8714           6              June        2016   
3        5     0.45 -383.0310          10           October        2015   
4        2     0.20    2.5164          10           October        2015   

   order Day of week  
0                  1  
1                  1  
2                  6  
3                  6  
4                  6  

[5 rows x 25 columns]

In [16]:
df.columns

Index(['Row ID', 'Order ID', 'Order Date', 'Ship Date', 'Ship Mode',
       'Customer ID', 'Customer Name', 'Segment', 'Country', 'City', 'State',
       'Postal Code', 'Region', 'Product ID', 'Category', 'Sub-Category',
       'Product Name', 'Sales', 'Quantity', 'Discount', 'Profit',
       'order month', 'order month_name', 'order year', 'order Day of week'],
      dtype='object')

# 1.MONTHLY SALES ANALYSIS

In [17]:
monthly_sales=df.groupby('order month')['Sales'].sum().reset_index()
monthly_sales

order month        Sales
0             1   94924.8356
1             2   59751.2514
2             3  205005.4888
3             4  137762.1286
4             5  155028.8117
5             6  152718.6793
6             7  147238.0970
7             8  159044.0630
8             9  307649.9457
9            10  200322.9847
10           11  352461.0710
11           12  325293.5035

In [18]:
fig= px.line(monthly_sales,x='order month',y='Sales', title='Monthly sales analysis')
fig.show()

Through this graph, we can visualize that Febraury(2)
had the lowest sales at approximately $59.7k, while November(11)had the highest sales at around 352.4k.

 # 2.SALES BY CATEGORY AND DETERMINE WHICH CATEGORY HAD THE HIGHEST AND LOWEST SALES


In [19]:
category_sales = df.groupby('Category')['Sales'].sum().reset_index()
category_sales

Category        Sales
0        Furniture  741999.7953
1  Office Supplies  719047.0320
2       Technology  836154.0330

In [20]:
fig=px.pie(category_sales,names='Category',values='Sales',title='Sales by Category',hole=0.2)
fig.update_traces(textposition='inside',textinfo='percent+label') #update_traces is used to modify the figure
fig.show()

This pie chart shows that out of the three product categories — Furniture, Technology, and Office Supplies — Technology made the most money, making up about 36.4% of total sales (around $836,154). On the other hand, Office Supplies made the least amount in sales.



# 3.SALES ANALYSIS NEEDS TO BE DONE BASED ON  SUB CATEGORIES

In [21]:
subcategory_sales= df.groupby(['Category','Sub-Category'])['Sales'].sum().reset_index()
subcategory_sales

Category Sub-Category        Sales
0         Furniture    Bookcases  114879.9963
1         Furniture       Chairs  328449.1030
2         Furniture  Furnishings   91705.1640
3         Furniture       Tables  206965.5320
4   Office Supplies   Appliances  107532.1610
5   Office Supplies          Art   27118.7920
6   Office Supplies      Binders  203412.7330
7   Office Supplies    Envelopes   16476.4020
8   Office Supplies    Fasteners    3024.2800
9   Office Supplies       Labels   12486.3120
10  Office Supplies        Paper   78479.2060
11  Office Supplies      Storage  223843.6080
12  Office Supplies     Supplies   46673.5380
13       Technology  Accessories  167380.3180
14       Technology      Copiers  149528.0300
15       Technology     Machines  189238.6310
16       Technology       Phones  330007.0540

In [22]:
fig=px.bar(subcategory_sales,x='Sub-Category',y='Sales',title= 'Sales by Sub-Category',color='Category')
fig.show()

# 4.ANALYZE THE MONTHLY PROFIT FROM SALES AND DETERMINE WHICH MONTH HAD THE HIGHEST PROFIT

In [23]:
df.head()

Row ID        Order ID Order Date  Ship Date       Ship Mode Customer ID  \
0       1  CA-2016-152156 2016-11-08 2016-11-11    Second Class    CG-12520   
1       2  CA-2016-152156 2016-11-08 2016-11-11    Second Class    CG-12520   
2       3  CA-2016-138688 2016-06-12 2016-06-16    Second Class    DV-13045   
3       4  US-2015-108966 2015-10-11 2015-10-18  Standard Class    SO-20335   
4       5  US-2015-108966 2015-10-11 2015-10-18  Standard Class    SO-20335   

     Customer Name    Segment        Country             City  ...  \
0      Claire Gute   Consumer  United States        Henderson  ...   
1      Claire Gute   Consumer  United States        Henderson  ...   
2  Darrin Van Huff  Corporate  United States      Los Angeles  ...   
3   Sean O'Donnell   Consumer  United States  Fort Lauderdale  ...   
4   Sean O'Donnell   Consumer  United States  Fort Lauderdale  ...   

  Sub-Category                                       Product Name     Sales  \
0    Bookcases                  Bush Somerset Collection Bookcase  261.9600   
1       Chairs  Hon Deluxe Fabric Upholstered Stacking Chairs,...  731.9400   
2       Labels  Self-Adhesive Address Labels for Typewriters b...   14.6200   
3       Tables      Bretford CR4500 Series Slim Rectangular Table  957.5775   
4      Storage                     Eldon Fold 'N Roll Cart System   22.3680   

  Quantity Discount    Profit order month  order month_name  order year  \
0        2     0.00   41.9136          11          November        2016   
1        3     0.00  219.5820          11          November        2016   
2        2     0.00    6.8714           6              June        2016   
3        5     0.45 -383.0310          10           October        2015   
4        2     0.20    2.5164          10           October        2015   

   order Day of week  
0                  1  
1                  1  
2                  6  
3                  6  
4                  6  

[5 rows x 25 columns]

In [24]:
monthly_profit= df.groupby('order month')['Profit'].sum().reset_index()
monthly_profit

order month      Profit
0             1   9134.4461
1             2  10294.6107
2             3  28594.6872
3             4  11587.4363
4             5  22411.3078
5             6  21285.7954
6             7  13832.6648
7             8  21776.9384
8             9  36857.4753
9            10  31784.0413
10           11  35468.4265
11           12  43369.1919

In [25]:
fig= px.line(monthly_profit,x='order month',y='Profit',title='Monthly Profit Analysis')
fig.show()

# 5.ANALYZE THE PROFIT BY CATEGORY AND SUB CATEGORY

In [26]:
profit_by_category= df.groupby('Category')['Profit'].sum().reset_index()
profit_by_category





Category       Profit
0        Furniture   18451.2728
1  Office Supplies  122490.8008
2       Technology  145454.9481

In [27]:
custom_colors = {
    'Furniture': 'green',
    'Technology': 'yellow',
    'Office Supplies': 'blue'}

fig= px.bar(profit_by_category,x='Category',y='Profit',title='Profit by Category',color='Category',
            color_discrete_map=custom_colors)  # Apply custom colors

fig.show()

In [28]:
profit_by_subcategory= df.groupby('Sub-Category')['Profit'].sum().reset_index()
profit_by_subcategory

Sub-Category      Profit
0   Accessories  41936.6357
1    Appliances  18138.0054
2           Art   6527.7870
3       Binders  30221.7633
4     Bookcases  -3472.5560
5        Chairs  26590.1663
6       Copiers  55617.8249
7     Envelopes   6964.1767
8     Fasteners    949.5182
9   Furnishings  13059.1436
10       Labels   5546.2540
11     Machines   3384.7569
12        Paper  34053.5693
13       Phones  44515.7306
14      Storage  21278.8264
15     Supplies  -1189.0995
16       Tables -17725.4811

In [29]:
fig= px.pie(profit_by_subcategory,names='Sub-Category',values='Profit',title='PROFIT BY SUB-CATEGORY')
fig.update_traces(textinfo='percent+label')
fig.show()

In [30]:
# To find Copiers is located in which category
df[df['Sub-Category']=='Copiers']['Category'].value_counts()

Category
Technology    68
Name: count, dtype: int64

# 6.SALES AND PROFIT - CUSTOMER SEGMENT

In [31]:
#We need to consider sales, profit and segment columns
df.head()

Row ID        Order ID Order Date  Ship Date       Ship Mode Customer ID  \
0       1  CA-2016-152156 2016-11-08 2016-11-11    Second Class    CG-12520   
1       2  CA-2016-152156 2016-11-08 2016-11-11    Second Class    CG-12520   
2       3  CA-2016-138688 2016-06-12 2016-06-16    Second Class    DV-13045   
3       4  US-2015-108966 2015-10-11 2015-10-18  Standard Class    SO-20335   
4       5  US-2015-108966 2015-10-11 2015-10-18  Standard Class    SO-20335   

     Customer Name    Segment        Country             City  ...  \
0      Claire Gute   Consumer  United States        Henderson  ...   
1      Claire Gute   Consumer  United States        Henderson  ...   
2  Darrin Van Huff  Corporate  United States      Los Angeles  ...   
3   Sean O'Donnell   Consumer  United States  Fort Lauderdale  ...   
4   Sean O'Donnell   Consumer  United States  Fort Lauderdale  ...   

  Sub-Category                                       Product Name     Sales  \
0    Bookcases                  Bush Somerset Collection Bookcase  261.9600   
1       Chairs  Hon Deluxe Fabric Upholstered Stacking Chairs,...  731.9400   
2       Labels  Self-Adhesive Address Labels for Typewriters b...   14.6200   
3       Tables      Bretford CR4500 Series Slim Rectangular Table  957.5775   
4      Storage                     Eldon Fold 'N Roll Cart System   22.3680   

  Quantity Discount    Profit order month  order month_name  order year  \
0        2     0.00   41.9136          11          November        2016   
1        3     0.00  219.5820          11          November        2016   
2        2     0.00    6.8714           6              June        2016   
3        5     0.45 -383.0310          10           October        2015   
4        2     0.20    2.5164          10           October        2015   

   order Day of week  
0                  1  
1                  1  
2                  6  
3                  6  
4                  6  

[5 rows x 25 columns]

In [32]:
df['Segment'].value_counts()

Segment
Consumer       5191
Corporate      3020
Home Office    1783
Name: count, dtype: int64

In [33]:
sp_segment= df.groupby('Segment').agg({'Sales':'sum','Profit':'sum'}).reset_index()
sp_segment

Segment         Sales       Profit
0     Consumer  1.161401e+06  134119.2092
1    Corporate  7.061464e+05   91979.1340
2  Home Office  4.296531e+05   60298.6785

In [34]:
fig= px.bar(sp_segment,x='Segment',y=['Sales','Profit'],barmode='group',title='Sales and Profit by Segment')
fig.show()

# 7.ANALYZE THE SALES TO PROFIT RATIO

In [35]:
sp_ratio= df.groupby('Segment').agg({'Sales':'sum','Profit':'sum'}).reset_index()
sp_ratio['sales_profit_ratio']=sp_ratio['Sales']/sp_ratio['Profit']
sp_ratio

Segment         Sales       Profit  sales_profit_ratio
0     Consumer  1.161401e+06  134119.2092            8.659471
1    Corporate  7.061464e+05   91979.1340            7.677245
2  Home Office  4.296531e+05   60298.6785            7.125416

# **CONCLUSION**

1)From the data visualization, I found that sales are highest in November, but the most profit is made in December. This might mean that even though people buy more in November, we earn more money per sale in December.

2)When we look at what people are buying, Technology products are the most popular, especially among regular consumers compared to office or corporate buyers.

3)Even though phones have the highest sales in the Technology category, we actually make more profit from copiers. Copiers give us both high sales and high profits, which makes them a very valuable product for the business